edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -980,3 +982,96 @@ def test_post_create_attempt(self, start_immediately, mock_update_attempt, mock_
 
         if start_immediately:
             mock_update_attempt.assert_called_once_with(mock_attempt_id, ExamAttemptStatus.started)
+
+
+class CourseExamAttemptViewTest(ExamsAPITestCase):
+    """
+    Tests CourseExamAttemptView
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = '11111111'
+
+        self.course_exam_config = CourseExamConfiguration.objects.create(
+            course_id=self.course_id,
+            provider=self.test_provider,
+            allow_opt_out=False
+        )
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+    def get_api(self, user, course_id, content_id):
+        """
+        Helper function to make patch request to the API
+        """
+
+        headers = self.build_jwt_headers(user)
+        url = reverse(
+            'api:v1:exams-course_attempt',
+            kwargs={'course_id': course_id, 'content_id': content_id}
+        )
+
+        return self.client.get(url, **headers)
+
+    def test_no_exam(self):
+        """
+        Test endpoint for a content ID that doesn't exist
+        """
+        response = self.get_api(self.user, self.course_id, 111111)
+        self.assertEqual(response.data['exam'], {})
+
+    def test_no_active_attempt(self):
+        """
+        Test endpoint for an existing exam, but no attempt for user
+        """
+        serialized_exam = ExamSerializer(self.exam).data
+
+        exam_type_class = get_exam_type(serialized_exam['exam_type'])
+
+        serialized_exam['type'] = serialized_exam['exam_type']
+        serialized_exam['is_proctored'] = exam_type_class.is_proctored
+        serialized_exam['is_practice_exam'] = exam_type_class.is_practice
+        serialized_exam['backend'] = serialized_exam['provider_name']
+        serialized_exam['attempt'] = {}
+
+        response = self.get_api(self.user, self.course_id, self.content_id)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNotNone(response.data['exam'])
```

Comment:
need to do more assertions here on the exam object depending on whether or not the exam gets a custom serializer